### PR TITLE
feat: add new service dynamic_cloud with resources vpc_project

### LINKED ISSUE #50: [Dynamic Cloud]: Manage Projects and user assignment
Create new Resources for Dynmaic Cloud Projects with
- Users (non Service users)
- Project settings

### PR DESCRIPTION
**Pull request by roehlc**
Fixes #50 

Todo:

- [x] implement Dynamic Cloud VPC Project creation / deletion
- [x] implement Dynamic Cloud VPC Project user management

### CODE REVIEW

**Comment by roehlc** [comment]
Needs rebase once #53 is merged

File: cancom/services/dynamic-cloud/resource_vpc_project.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	tflog.Info(ctx, "Deleting VPC Project")
+	var diags diag.Diagnostics
+	vpcProjectShortid := d.Id()
+
+	err = (*client_dynamiccloud.Client)(c).DeleteVpcProject(vpcProjectShortid)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutDelete)-time.Minute, func() *resource.RetryError {
```
Comment (atddo) [inline cancom/services/dynamic-cloud/resource_vpc_project.go:247]: Tested, got an error.
cancom_dynamic_cloud_vpc_project.name: Still destroying... [id=9SHbs566yAS6fhjqfhZpoU, 14m1s elapsed]
╷
│ Error: VPC Project is still transitioning with phase 'Deleting'
│ 


Maybe adjust the timeout.  After 16 min, the backend reports stillt the status 
phase": "Deleting",

**Review by atddo — CHANGES_REQUESTED** [review]
Thank you for your implementation.
Tested the resource:
Create, Read, Update is fine.
Delete runs into timeout (see comment).

File: cancom/services/dynamic-cloud/resource_vpc_project.go (hunk 1 above)
Comment (roehlc) [reply cancom/services/dynamic-cloud/resource_vpc_project.go:247]: Thanks you or testing the changes in this PR.
This was not a bug in the provider. We changed how the backend handles deletion and forget to change a reference on the callback telling us the deletion finished.
It's fixed now.

**Review by atddo — APPROVED** [review]
Reviewed

**Pull request merged**